Cloudwatch sort_frame crash

**What happened**:
Grafana workspace crashed when attempting to use CloudWatch [sort_frame](https://github.com/grafana/grafana/blob/master/pkg/tsdb/cloudwatch/sort_frame.go#L40)

application logs show:
```
panic: runtime error: invalid memory address or nil pointer dereference,
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a0e825]
goroutine 2067409 [running]:
github.com/grafana/grafana/pkg/tsdb/cloudwatch.ByTime.Less(0x2323181, 0x16, 0xc002545980, 0x3, 0x3, 0x0, 0x0, 0xc001857da0, 0x0, 0x185, ...)
\t/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/sort_frame.go:40 +0x125
sort.doPivot(0x268ef80, 0xc0024f41c0, 0x0, 0x307, 0x3, 0x3)
\t/usr/local/go/src/sort/sort.go:121 +0x537
sort.quickSort(0x268ef80, 0xc0024f41c0, 0x0, 0x307, 0x14)
\t/usr/local/go/src/sort/sort.go:190 +0x9a
sort.Sort(0x268ef80, 0xc0024f41c0)
\t/usr/local/go/src/sort/sort.go:218 +0x7a
github.com/grafana/grafana/pkg/tsdb/cloudwatch.logsResultsToDataframes(0xc0024beff0, 0x2698bc0, 0xc0011e1540, 0x26cd260)
\t/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_query.go:142 +0xf0a
\t/go/pkg/mod/golang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:54 +0x66
sort.quickSort(0x268ef80, 0xc0024f41c0, 0x0, 0x307, 0x14
\t/usr/local/go/src/sort/sort.go:190 +0x9a
sort.Sort(0x268ef80, 0xc0024f41c0)
\t/usr/local/go/src/sort/sort.go:218 +0x7a
github.com/grafana/grafana/pkg/tsdb/cloudwatch.logsResultsToDataframes(0xc0024beff0, 0x2698bc0, 0xc0011e1540, 0x26cd260)
\t/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_query.go:142 +0xf0a
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*cloudWatchExecutor).handleGetQueryResults(0xc0011e1500, 0x2698bc0, 0xc0011e1540, 0x26cd260, 0xc000010b50, 0xc002065370, 0x35c01a8, 0x1, 0x41cf25, 0x7f32bca9bd00, ...)
\t/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_actions.go:301 +0x86
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*cloudWatchExecutor).executeLogAction(0xc0011e1500, 0x2698bc0, 0xc0011e1540, 0xc0011e1480, 0xc0011e14c0, 0x4, 0xc0010faf08, 0x45255f)
\t/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_actions.go:102 +0x4a5
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*cloudWatchExecutor).executeLogActions.func1(0xc006fbe0c0, 0x2661201)
\t/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/log_actions.go:25 +0x6e
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0024bee40, 0xc0011e1580)
\t/go/pkg/mod/golang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
\t/go/pkg/mod/golang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:54 +0x66


t=2021-03-05T02:59:13+0000 lvl=info msg="Starting Grafana" logger=server version=7.3.5
--



```

**What you expected to happen**:
* Null checks. 
* Error handling does not result in workspace crashes

**Environment**:
- Grafana version: 7.3.5
- Data source type & version: CloudWatch
- OS Grafana is installed on: /
- User OS & Browser: unavailable 
- Grafana plugins: /
- Others: /

--- a/pkg/tsdb/cloudwatch/sort_frame.go
+++ b/pkg/tsdb/cloudwatch/sort_frame.go
@@ -37,5 +37,13 @@ func (a ByTime) Less(i, j int) bool {
 		return false
 	}
 
+	if timeField.At(i).(*time.Time) == nil {
+		return false
+	}
+
+	if timeField.At(j).(*time.Time) == nil {
+		return true
+	}
+
 	return (timeField.At(i).(*time.Time)).Before(*timeField.At(j).(*time.Time))
 }

--- a/pkg/tsdb/cloudwatch/sort_frame_test.go
+++ b/pkg/tsdb/cloudwatch/sort_frame_test.go
@@ -12,46 +12,77 @@ import (
 )
 
 func TestFrameSort(t *testing.T) {
-	timeA, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 17:04:05.000")
-	timeB, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 16:04:05.000")
-	timeC, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 15:04:05.000")
-	timeVals := []*time.Time{
-		&timeA, &timeB, &timeC,
-	}
-	timeField := data.NewField("@timestamp", nil, timeVals)
+	t.Run("sort simple frame", func(t *testing.T) {
+		timeA, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 17:04:05.000")
+		timeB, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 16:04:05.000")
+		timeC, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 15:04:05.000")
+		timeVals := []*time.Time{
+			&timeA, &timeB, &timeC,
+		}
+		timeField := data.NewField("@timestamp", nil, timeVals)
 
-	stringField := data.NewField("line", nil, []*string{
-		aws.String("test message 1"),
-		aws.String("test message 2"),
-		aws.String("test message 3"),
+		stringField := data.NewField("line", nil, []*string{
+			aws.String("test message 1"),
+			aws.String("test message 2"),
+			aws.String("test message 3"),
+		})
+
+		numberField := data.NewField("nums", nil, []*float64{
+			aws.Float64(20.0),
+			aws.Float64(50.0),
+			aws.Float64(17.0),
+		})
+
+		expectedDataframe := &data.Frame{
+			Name: "CloudWatchLogsResponse",
+			Fields: []*data.Field{
+				timeField,
+				stringField,
+				numberField,
+			},
+		}
+
+		sort.Sort(ByTime(*expectedDataframe))
+
+		for i := 1; i < timeField.Len(); i++ {
+			assert.True(t, timeField.At(i).(*time.Time).After(*(timeField.At(i - 1).(*time.Time))))
+		}
+
+		assert.Equal(t, *stringField.At(0).(*string), "test message 3")
+		assert.Equal(t, *stringField.At(1).(*string), "test message 2")
+		assert.Equal(t, *stringField.At(2).(*string), "test message 1")
+
+		assert.Equal(t, *numberField.At(0).(*float64), 17.0)
+		assert.Equal(t, *numberField.At(1).(*float64), 50.0)
+		assert.Equal(t, *numberField.At(2).(*float64), 20.0)
 	})
 
-	numberField := data.NewField("nums", nil, []*float64{
-		aws.Float64(20.0),
-		aws.Float64(50.0),
-		aws.Float64(17.0),
+	t.Run("sort with nil", func(t *testing.T) {
+		timeA, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 17:04:05.000")
+		timeB, _ := time.Parse("2006-01-02 15:04:05.000", "2020-03-02 16:04:05.000")
+		timeVals := []*time.Time{
+			&timeA, &timeB, nil,
+		}
+		timeField := data.NewField("@timestamp", nil, timeVals)
+
+		stringField := data.NewField("line", nil, []*string{
+			aws.String("test message 1"),
+			aws.String("test message 2"),
+			aws.String("test message 3"),
+		})
+
+		frame := &data.Frame{
+			Name: "CloudWatchLogsResponse",
+			Fields: []*data.Field{
+				timeField,
+				stringField,
+			},
+		}
+
+		sort.Sort(ByTime(*frame))
+
+		assert.Equal(t, *stringField.At(0).(*string), "test message 2")
+		assert.Equal(t, *stringField.At(1).(*string), "test message 1")
+		assert.Equal(t, *stringField.At(2).(*string), "test message 3")
 	})
-
-	expectedDataframe := &data.Frame{
-		Name: "CloudWatchLogsResponse",
-		Fields: []*data.Field{
-			timeField,
-			stringField,
-			numberField,
-		},
-	}
-
-	sort.Sort(ByTime(*expectedDataframe))
-
-	for i := 1; i < timeField.Len(); i++ {
-		assert.True(t, timeField.At(i).(*time.Time).After(*(timeField.At(i - 1).(*time.Time))))
-	}
-
-	assert.Equal(t, *stringField.At(0).(*string), "test message 3")
-	assert.Equal(t, *stringField.At(1).(*string), "test message 2")
-	assert.Equal(t, *stringField.At(2).(*string), "test message 1")
-
-	assert.Equal(t, *numberField.At(0).(*float64), 17.0)
-	assert.Equal(t, *numberField.At(1).(*float64), 50.0)
-	assert.Equal(t, *numberField.At(2).(*float64), 20.0)
 }

